common/commands/src/validator/transactions/query_transactions.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use clap::Parser;
+use cosmrs::rpc::query::Query;
+use log::error;
+use serde_json::json;
+
+use crate::context::QueryClient;
+
+#[derive(Debug, Parser)]
+pub struct Args {
+    #[clap(value_parser)]
+    #[clap(help = "The query to execute")]
+    pub query: String,
+}
+
+pub async fn query(args: Args, client: &QueryClient) {
+    match Query::from_str(&args.query) {
+        Ok(query) => match client.search_tx(query).await {
+            Ok(res) => {
+                println!("{}", json!(res))
+            }
+            Err(e) => {
+                error!("{:?}", e);
+            }
+        },
+        Err(e) => error!("{:?}", e),
```

Comment:
Seeing this match pattern with errors everywhere (i.e. `Err(e) => error!("{:?}", e),`) I wonder if it would make sense to make all commands return a `Result` (maybe `anyhow::Result`) propagate those errors and make the top-level function print the errors to terminal?

Reply:
I started down that path, and then encountered all the old commands from the `validator-client-scripts`. Was thinking later the commands could be refactored to delegate handling of outputs and errors to something higher up.